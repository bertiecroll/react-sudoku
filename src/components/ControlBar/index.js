export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
