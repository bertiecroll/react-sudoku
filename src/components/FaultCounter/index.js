export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
