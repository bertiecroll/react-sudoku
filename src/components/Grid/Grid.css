.Grid {
  width: 27em;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  flex-direction: column-reverse;
}

.GridRow {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  flex-direction: row;
}

.GridRow:nth-last-child(3n) {
  border-bottom: 5px solid #777;
}

.GridRow:nth-last-child(3n + 1) .GridCell {
  padding-top: .25em;
}

.GridRow:first-child {
  border-bottom: none;
}
