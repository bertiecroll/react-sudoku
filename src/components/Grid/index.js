export { Grid } from './Grid'
export { GridRow } from './GridRow'
