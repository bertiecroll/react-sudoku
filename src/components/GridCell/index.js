export { GridCell } from './GridCell'
export { EmptyGridCell } from './EmptyGridCell'
export { CompleteGridCell } from './CompleteGridCell'
export { default } from './GridCell'
