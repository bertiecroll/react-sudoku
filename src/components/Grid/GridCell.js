import React from 'react'
import PropTypes from 'prop-types'
import './Grid.css'

export const GridCell = (props) => {
  return (
    <div className="GridCell">
      { props.number }
    </div>
  )
}

GridCell.propTypes = {
  number: PropTypes.number.isRequired
}
