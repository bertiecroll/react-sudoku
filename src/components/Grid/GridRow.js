import React from 'react'
import PropTypes from 'prop-types'
import './Grid.css'
import { GridCell } from './GridCell'

export const GridRow = (props) => {
  return (
    <div className="GridRow">
      {props.row.map((number, index) => <GridCell key={index} number={number} selected={ props.selected && index === props.selectedCell.column } />)}
    </div>
  )
}

GridRow.propTypes = {
  row: PropTypes.array.isRequired
}
