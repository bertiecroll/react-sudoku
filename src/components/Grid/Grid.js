import React from 'react'
import PropTypes from 'prop-types'
import './Grid.css'
import { GridRow } from './GridRow'
import { mapWithIndex } from 'lib/utils'

export const Grid = (props) => {
  return (
    <div className="Grid">
      { mapWithIndex((cells, key) => <GridRow key={key} cells={cells} />)(props.cellsByRow) }
    </div>
  )
}

Grid.propTypes = {
  cellsByRow: PropTypes.object.isRequired,
}
