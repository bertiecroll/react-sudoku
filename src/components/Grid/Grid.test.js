import React from 'react'
import { shallow } from 'enzyme'
import { Grid } from 'components/Grid'

describe('Grid', () => {
  it('renders a GridRow for each props.cellsByRow, passing key and cells to each GridRow', () => {
    const cellsByRow = {
      0: [{ id: '0,0', value: 1 }, { id: '1,0', value: 2 }, { id: '2,0', value: 3 }],
      1: [{ id: '0,1', value: 4 }, { id: '1,1', value: 5 }, { id: '2,1', value: 6 }],
      2: [{ id: '0,2', value: 7 }, { id: '1,2', value: 8 }, { id: '2,2', value: 9 }],
    }
    const wrapper = shallow(<Grid cellsByRow={cellsByRow} />)
    expect(wrapper).toMatchSnapshot()
  })
})
