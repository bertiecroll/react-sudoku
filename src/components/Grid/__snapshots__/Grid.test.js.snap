// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders a GridRow for each props.cellsByRow, passing key and cells to each GridRow 1`] = `
<div
  className="Grid"
>
  <GridRow
    cells={
      Array [
        Object {
          "id": "0,0",
          "value": 1,
        },
        Object {
          "id": "1,0",
          "value": 2,
        },
        Object {
          "id": "2,0",
          "value": 3,
        },
      ]
    }
    key="0"
  />
  <GridRow
    cells={
      Array [
        Object {
          "id": "0,1",
          "value": 4,
        },
        Object {
          "id": "1,1",
          "value": 5,
        },
        Object {
          "id": "2,1",
          "value": 6,
        },
      ]
    }
    key="1"
  />
  <GridRow
    cells={
      Array [
        Object {
          "id": "0,2",
          "value": 7,
        },
        Object {
          "id": "1,2",
          "value": 8,
        },
        Object {
          "id": "2,2",
          "value": 9,
        },
      ]
    }
    key="2"
  />
</div>
`;
